.github/workflows/tests-ubuntu.yml

```diff
@@ -48,8 +47,30 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.10']
-        tox-job: ["mypy", "docs", "linters", "types"]
+        python-version: ["3.10"]
+        tox-job: ["mypy", "linters", "types"]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install tox
+    - name: tox
+      run: |
+        tox -e ${{ matrix.tox-job }}
+
+  docs:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7"]
+        tox-job: ["docs"]
```

Comment:
Not the cleanest solution as there's currently no way for GitHub actions to reuse steps ([reference](https://stackoverflow.com/a/61488170)). If there is, we can make it much shorter.